dxr/plugins/python/indexers.py

```diff
@@ -35,19 +42,115 @@
         'properties': {
             'py_type': NEEDLE,
             'py_function': NEEDLE,
+            'py_derived': NEEDLE,
+            'py_bases': NEEDLE,
         },
     },
 }
 
 
+class TreeToIndex(TreeToIndexBase):
+    def __init__(self, *args, **kwargs):
+        super(TreeToIndex, self).__init__(*args, **kwargs)
+
+        self.python_path = getattr(self.tree, 'plugin_python_path', self.tree.source_folder)
+        self.class_bases = {}
+        self.derived_classes = defaultdict(list)
+        self.names = {}
+
+    @property
+    def unignored_files(self):
+        return unignored(self.tree.source_folder, self.tree.ignore_paths,
+                         self.tree.ignore_filenames)
+
+    def post_build(self):
+        for path in self.unignored_files:
+            if is_interesting(path):
+                self.analyze(path)
+
+        # Once we've got all the base classes, store the opposite
+        # direction: derived classes!
+        for class_name, bases in self.class_bases.items():
+            for base_name in bases:
+                base_name = normalize_name(self.names, base_name)
+                self.derived_classes[base_name].append(class_name)
+
+    def analyze(self, path):
+        """Parse a Python file and analyze it for import and class data
```

Comment:
Can you mention that this stows the results in instance vars? It'll make a little faster (for me) to reason about.